### ISSUE TITLE
size() is inconsistent with shape() for 1D slice view

### ISSUE THREAD

**Issue by stuarteberg** [QuantStack/xtensor-python#119]
If I'm understanding the docs for [`xarray::size()`](https://xtensor.readthedocs.io/en/latest/api/xcontainer.html#_CPPv2NK2xt10xcontainer4sizeEv) correctly, then in the 1D case, `size()` and `shape()[0]` should always be identical.  Is that correct?

In the following `pyarray` test case, `size()` is not consistent with its `shape()`.  Notice that xtensor thinks that `col0` has a size is `10` even though its shape is only `(5,)`

I discovered this using the latest releases of `xtensor` and `xtensor-python`, but I can reproduce the behavior using the latest master branches.

Test case:

```python
import numpy as np
from example import print_xt_info

rows = np.zeros((5,2), dtype=np.uint32)
print(f"rows:\t size: {rows.size} shape: {rows.shape} strides: {rows.strides}")
print_xt_info(rows)

col0 = rows[:,0]
print(f"col0:\t size: {col0.size} shape: {col0.shape} strides: {col0.strides}")
print_xt_info(col0)
```

Output:

```
rows:	 size: 10 shape: (5, 2) strides: (8, 4)
xt_a:	 size: 10 shape: [5,2,] strides: [2,1,]

col0:	 size: 5 shape: (5,) strides: (8,)
xt_a:	 size: 10 shape: [5,] strides: [2,]
```

<details>
<summary>Click for C++ module</summary>

```c++
#include <iostream>
#include <cstdint>
#include "pybind11/pybind11.h"

#define FORCE_IMPORT_ARRAY
#include "xtensor-python/pyarray.hpp"

namespace py = pybind11;
using std::uint32_t;

template <class T>
std::string seq_to_string(T v)
{
    std::ostringstream ss;
    ss << "[";
    for ( auto x : v )
        ss << x << ",";
    ss << "]";
    return ss.str();
}

inline void print_xt_info(xt::pyarray<uint32_t> const & xa)
{
    std::cout << "xt_a:\t"
              << " size: " << xa.size()
              << " shape: " << seq_to_string(xa.shape())
              << " strides: " << seq_to_string(xa.strides())
              << std::endl << std::endl;
}

PYBIND11_PLUGIN(example)
{
    xt::import_numpy();
    py::module m("example", "");
    m.def("print_xt_info", print_xt_info);
    return m.ptr();
}
```

</details>

**Comment by wolfv** [comment]
There was a similar issue in xtensor-blas. Currently size returns the size of the underlying buffer. 
But I agree that I think it would make more sense to return the total number of elements.

**Comment by SylvainCorlay** [comment]
Grumbles, I thought that we had changed this behavior to be like numpy, but it seems not.

**Comment by SylvainCorlay** [comment]
@stuarteberg thanks for reporting! Btw, you can make quick xtensor experiments in the xeus notebook. There is a `launch binder` badge on the top of the readme!

**Comment by stuarteberg** [comment]
Thanks for the tip, @SylvainCorlay.  Man, that's so freakin' cool.

Regarding this `size()` issue, FWIW: I can't reproduce it in pure C++.  (Not in the notebook, anyway.)  

I guess this issue only applies to `xtensor-python`.

```c++
#include <iostream>
#include "xtensor/xarray.hpp"
#include "xtensor/xio.hpp"

using std::uint32_t;
{
    xt::xarray<uint32_t> rows = xt::zeros<uint32_t>({5,2});
    auto col0 = xt::view(rows, xt::all(), 0);
    std::cout << col0.size() << std::endl;
}
```

```
5
```

**Comment by SylvainCorlay** [comment]
Regarding the `size()` issue, we agree that the behavior is wrong. We will look into it in the next couple of days for `0.13.0`.

**Comment by wolfv** [comment]
This'll also fix https://github.com/QuantStack/xtensor-blas/issues/32

**Comment by wolfv** [comment]
just pushed a fix to xtensor: https://github.com/QuantStack/xtensor/pull/502

If the lights turn green, I think we can merge it today.

**Comment by SylvainCorlay** [comment]
Fixed in QuantStack/xtensor#502

**Comment by SylvainCorlay** [comment]
The next version of xtensor-python will depend on xtensor 0.13 which has this fix.
